### PR TITLE
Only build and publish new ROCK version if rockfile.yaml changed

### PR DESCRIPTION
**Pull request by maci3jka**

### CODE REVIEW

**Comment by maci3jka** [comment]
branch run https://github.com/canonical/fluent-bit-rock/actions/runs/10245631270
not triggered pr https://github.com/canonical/fluent-bit-rock/pull/9/files
triggered build for one rock https://github.com/canonical/fluent-bit-rock/pull/10

File: .github/workflows/build_rocks.yaml [hunk 1]
```diff
@@ -122,43 +132,60 @@ jobs:
...
+
+
+            if (${{ github.event_name == 'pull_request' }}  && changesPaths.includes(rockcraftFile)){
+                    changedMetas.push(meta)
+            } else if (${{ github.event_name != 'pull_request' }}) {
+                    changedMetas.push(meta)
+            }
               }
               rockPaths.push(rockPath)
             }
```
Comment (bschimke95) [inline .github/workflows/build_rocks.yaml:175]: Please properly intend

File: .github/workflows/build_rocks.yaml [hunk 2]
```diff
@@ -122,43 +132,60 @@ jobs:
...
             core.setOutput('rock-metas', JSON.stringify(rockMetas))
+            core.setOutput('changed-metas', JSON.stringify(changedMetas))
             core.setOutput('rock-paths', JSON.stringify(rockPaths))
             core.setOutput('images', JSON.stringify(images))
+            core.info(`changedMetas: ${changedMetas}`) // todo remove after development
```
Comment (bschimke95) [inline .github/workflows/build_rocks.yaml:180]: remove

**Review by bschimke95 — COMMENTED** [review]
@maci3jka I'm not sure this will work like this.

If you only build the Rocks that have changed then the integration tests down the line will likely fail.

It is fine to *build* those ROCKs. Just don't *publish* them.

Looking at the linked test run they would still be published:
https://github.com/canonical/fluent-bit-rock/actions/runs/10278858107/job/28443209885?pr=9

File: .github/workflows/build_rocks.yaml [hunk 3]
```diff
@@ -96,69 +107,85 @@ jobs:
...
             core.setOutput('rock-metas', JSON.stringify(rockMetas))
+            core.setOutput('changed-metas', JSON.stringify(changedMetas))
             core.setOutput('rock-paths', JSON.stringify(rockPaths))
             core.setOutput('images', JSON.stringify(images))
 
   build-rocks:
     name: Build rock
     needs: [get-rocks]
-    if: ${{ needs.get-rocks.outputs.rock-metas != '[]' }}
+    if: ${{ needs.get-rocks.outputs.changed-metas != '[]' }}
```
Comment (bschimke95) [inline .github/workflows/build_rocks.yaml:185]: please rename this to something like `changed-rock-metas` to indicate to which kind of metas you are referring.

**Pull request merged**